my-app/src/components/Header/index.tsx

```diff
@@ -0,0 +1,303 @@
+                margin: '5px',
+                alignItems: 'center',
+              }}
+            >
+              <Tooltip title={t('main.signIn')}>
+                <Link to="sign-in">
+                  <PersonIcon fontSize="large" color="secondary" />
+                </Link>
+              </Tooltip>
+              <Tooltip title={t('main.signUp')}>
+                <Link to="sign-up">
+                  <PersonAddAlt1Icon fontSize="large" color="secondary" />
+                </Link>
+              </Tooltip>
+            </Box>
+          )}
+          {userId ? (
+            <Box sx={{ flexGrow: 0, display: { xs: 'flex', sm: 'none' } }}>
+              <Tooltip title="Open settings">
+                <IconButton onClick={handleOpenUserMenu} sx={{ p: 0 }}>
+                  <PersonIcon fontSize="large" />
+                </IconButton>
+              </Tooltip>
+              <Menu
+                sx={{ mt: '45px' }}
+                id="menu-appbar"
+                anchorEl={anchorElUser}
+                anchorOrigin={{
+                  vertical: 'top',
+                  horizontal: 'right',
+                }}
+                keepMounted
+                transformOrigin={{
+                  vertical: 'top',
+                  horizontal: 'right',
+                }}
+                open={Boolean(anchorElUser)}
+                onClose={handleCloseUserMenu}
+              >
+                <MenuItem>
+                  <Typography
+                    sx={{
+                      textAlign: 'center',
+                      textDecoration: 'none',
+                      color: theme.palette.secondary.main,
+                      padding: 0,
+                    }}
+                    onClick={() => dispatch(logout())}
+                  >
+                    {t('main.LOGOUT')}
+                  </Typography>
+                </MenuItem>
+                <MenuItem>
+                  <Link to="/boards">
+                    <Typography
+                      sx={{
+                        textAlign: 'center',
+                        textDecoration: 'none',
+                        color: theme.palette.secondary.main,
+                      }}
+                    >
+                      <Trans i18nKey="header.toMain"></Trans>
+                    </Typography>
+                  </Link>
+                </MenuItem>
+                <MenuItem>
+                  <Link to="/user-page">
+                    <Typography
+                      sx={{
+                        textAlign: 'center',
+                        textDecoration: 'none',
+                        color: theme.palette.secondary.main,
+                      }}
+                    >
+                      <Trans i18nKey="header.edit"></Trans>
+                    </Typography>
+                  </Link>
+                </MenuItem>
+              </Menu>
+            </Box>
+          ) : (
+            <></>
```

Comment:
You can write null for render empty element

Reply:
Thank you!

Reply:
for such cases use {condition && <React.Component/>}